Adding support for the minghe mhs5200

CatherineH opened the pull request.

scasagrande commented on instruments/minghe/mhs5200a.py:
```diff
@@ -0,0 +1,226 @@
+# CLASSES #####################################################################
+
+
+class MHS5200(Instrument):
```
Since this instrument is a function generator, this should probably inherit from the `FunctionGenerator` ABC

scasagrande commented on instruments/minghe/mhs5200a.py:
```diff
@@ -0,0 +1,226 @@
+
+    # INNER CLASSES #
+
+    class Channel(object):
```
Or instead of the main instrument class inheriting from `FunctionGenerator`, perhaps this one would be a better candidate.

CatherineH replied:
I'm not sure how this would work. If there are multiple channels, would each need an open file handle? I think the system would complain about multiple instances accessing the same file.

CatherineH replied:
also, why does amplitude have a _set_amplitude and a _get_amplitude function, but not _set_frequency and _get_frequency?

scasagrande replied:
`_get_amplitude_` and `_set_amplitude_` are there because we have a concrete implementation for `amplitude` in `FunctionGenerator`. This implementation is what gets called first, and handles conversion for volts vpp vs rms, and dBm. For setting, the numerical value, and the units, are then passed to `_set_amplitude_` so that you can then set things correctly on the instrument.

This lets you do this:

```python
>>> inst.amplitude = (5 * pq.V, inst.VoltageMode.rms)
>>> inst.amplitude = (1 * pq.V, inst.VoltageMode.peak_to_peak)
>>> inst.amplitude = 7 * pq.V  # assumes vpp
>>> inst.amplitude = 1 * ik.units.dBm
```

scasagrande commented on instruments/minghe/mhs5200a.py:
```diff
@@ -0,0 +1,226 @@
+
+        @amplitude.setter
+        def amplitude(self, new_val):
+            new_val = 100*assume_units(new_val, pq.V).rescale(pq.V).magnitude
```
Why do we need to multiply by 100? What a strange number system this instrument uses.

scasagrande commented on instruments/minghe/mhs5200a.py:
```diff
@@ -0,0 +1,226 @@
+            """
+            Gets/Sets the enable state of this channel.
+
+            :param bool new_val: the enable state
```
Replace with 
```
:type: `bool`
```

scasagrande commented on instruments/minghe/mhs5200a.py:
```diff
@@ -0,0 +1,226 @@
+
+            :units: As specified (if a `~quantities.Quantity`) or assumed to be
+            of fraction.
+            :type: `~quantities.Quantity`
```
Docstring here says that the offset is a `quantities.Quantity` unitful object, but the getter does not return units, and the setter does not handle unitful values.

scasagrande commented on instruments/minghe/mhs5200a.py:
```diff
@@ -0,0 +1,226 @@
+
+            :units: As specified (if a `~quantities.Quantity`) or assumed to be
+            of degrees.
+            :type: `~quantities.Quantity`
```
This property should use `pq.deg` as the units for getting and setting.

scasagrande commented on instruments/minghe/mhs5200a.py:
```diff
@@ -0,0 +1,226 @@
+            response = self._mhs.query(query)
+
+        @property
+        def wave_type(self):
```
`FunctionGenerator` ABC calls this `function`

CatherineH commented:
This has been tested with the function generator and it works.

scasagrande commented:
Uh, what's going on with that last commit?

I'm currently working (working from commit fa654cc) on trying to come up with a nice solution to the function generator base class. I didn't write it expecting multiple channel FGs, so I'm just tweaking things.

CatherineH commented:
I implemented the duty cycle wrong, and in trying to bring that change in, reverted some other changes. I'm going to roll it back to fa654cc and try to add it to that.
